packages/playwright-core/src/client/elementHandle.ts

```diff
@@ -286,19 +283,17 @@ export async function convertInputFiles(files: string | FilePayload | string[] |
     return { localPaths: items.map(f => path.resolve(f as string)) as string[] };
   }
 
-  const filePayloads: SetInputFilesFiles = await Promise.all(items.map(async item => {
-    if (typeof item === 'string') {
-      return {
-        name: path.basename(item),
-        buffer: await fs.promises.readFile(item)
-      };
-    } else {
-      return {
-        name: item.name,
-        mimeType: item.mimeType,
-        buffer: item.buffer,
-      };
-    }
+  const sizeLimit = 50 * 1024 * 1024;
+  const totalBufferSizeExceedsLimit = items.reduce((size, item) => size + ((typeof item === 'object' && item.buffer) ? item.buffer.byteLength : 0), 0) > sizeLimit;
```

Comment:
If you cast `items` to `FilePayload[]` before this line, there is no need for `typeof` and other checks.